app/src/main/java/org/android/go/sopt/presentation/auth/SignUpViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package org.android.go.sopt.presentation.auth
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import org.android.go.sopt.data.remote.SignUpRequestDTO
+import org.android.go.sopt.data.remote.SignUpResponseDTO
+import org.android.go.sopt.module.AuthServicePool
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.util.regex.Pattern
+
+class SignUpViewModel : ViewModel() {
+    private val _signUpResult: MutableLiveData<SignUpResponseDTO> = MutableLiveData()
+    val signUpResult: LiveData<SignUpResponseDTO> = _signUpResult
+
+    private val _errorResult: MutableLiveData<String> = MutableLiveData()
+    val errorResult: LiveData<String> = _errorResult
+
+    val idText: MutableLiveData<String> = MutableLiveData("")
+    val pwText: MutableLiveData<String> = MutableLiveData("")
+    val nameText: MutableLiveData<String> = MutableLiveData("")
+    val skillText: MutableLiveData<String> = MutableLiveData("")
+
+    val isIdValid: LiveData<Boolean> = Transformations.map(idText) { checkIdValid(it) }
+    val isPwValid: LiveData<Boolean> = Transformations.map(pwText) { checkPwValid(it) }
+
+    val isButtonValid: MutableLiveData<Boolean> = MutableLiveData(false)
+
+    fun signUp(id: String, password: String, name: String, skill: String) {
```

Comment:
양방향 데이터 바인딩을 통해 입력한 값이 뷰모델 변수로 전달되기 때문에 굳이 매개변수를 전달하지 않아도 됩니당
```suggestion
fun signUp() {
        AuthServicePool.authService.signUp(
            SignUpRequestDTO(
            idText, pwText, nameText, skillText
            )
        )
```

Reply:
오와 신기하네요 수정할게요 !